cli/azd/pkg/commands/pipeline/github_provider.go

```diff
@@ -358,17 +364,70 @@ func (p *GitHubCiProvider) configureConnection(
 	repoDetails *gitRepositoryDetails,
 	infraOptions provisioning.Options,
 	credentials json.RawMessage,
+	authType PipelineAuthType,
 	console input.Console) error {
 
+	// Oidc + Terraform is not a supported combination
+	if infraOptions.Provider == provisioning.Terraform {
+		// Throw error if Oidc is explicitly requested
+		if authType == AuthTypeOidc {
+			return fmt.Errorf(
+				"Terraform does not support OIDC authentication. Service Principal with client ID and client secret must be used. %w",
+				ErrAuthNotSupported,
+			)
+		}
+
+		// If not explicitly set, show warning
+		console.Message(
+			ctx,
+			output.WithWarningFormat(
+				//nolint:lll
+				"WARNING: Terraform provisioning does not support OIDC authentication, defaulting to Service Principal with client ID and client secret.\n",
+			),
```

Comment:
This warning will always be printed for terraform. Even if setting `--auth-type client-secret`.  Is that what we want?